### PR TITLE
Run custom plugins immediately on startup

### PR DESCRIPTION
**Pull request by yguo0905**
`time.Ticker` doesn't fire instantly so the custom plugins are not run immediately on NPD startup. Some plugins may be configured with long pull interval.

We'd like to see their first run when the cluster is just created, so changing it accordingly.

Following the advice on https://github.com/golang/go/issues/17601#issuecomment-319109006.

/assign @Random-Liu 
/assign @wangzhen127

### CODE REVIEW

File: pkg/custompluginmonitor/plugin/plugin.go [hunk 1]
```diff
@@ -55,41 +55,53 @@ func (p *Plugin) GetResultChan() <-chan cpmtypes.Result {
 func (p *Plugin) Run() {
 	runTicker := time.NewTicker(*p.config.PluginGlobalConfig.InvokeInterval)
```
Comment (xueweiz) [inline pkg/custompluginmonitor/plugin/plugin.go]: I think it's good to also clean up this code a little bit.
We could add a `defer runTicker.Stop()` line after the initialization of runTicker:
`runTicker := time.NewTicker(*p.config.PluginGlobalConfig.InvokeInterval)`

File: pkg/custompluginmonitor/plugin/plugin.go [hunk 2]
```diff
@@ -55,41 +55,53 @@ func (p *Plugin) GetResultChan() <-chan cpmtypes.Result {
...
+		glog.Info("Finish running custom plugins")
+	}
 
-					glog.Infof("Add check result %+v for rule %+v", result, rule)
-				}(rule)
-			}
+	select {
+	case <-p.tomb.Stopping():
+		glog.Info("Stopping plugin execution")
+		p.tomb.Done()
```
Comment (xueweiz) [inline pkg/custompluginmonitor/plugin/plugin.go]: Perhaps it's better to just move this one `p.tomb.Done()` and the one below to the top, as a `defer p.tomb.Done()`

File: pkg/custompluginmonitor/plugin/plugin.go [hunk 3]
```diff
@@ -55,41 +55,53 @@ func (p *Plugin) GetResultChan() <-chan cpmtypes.Result {
...
 
-			p.Wait()
-			glog.Info("Finish running custom plugins")
+	for {
+		select {
+		case <-runTicker.C:
+			runner()
 		case <-p.tomb.Stopping():
 			glog.Info("Stopping plugin execution")
 			p.tomb.Done()
```
Comment (xueweiz) [inline pkg/custompluginmonitor/plugin/plugin.go]: This is the second `p.tomb.Done()` I'd prefer to be `defer`-ed

File: pkg/custompluginmonitor/plugin/plugin.go (hunk 1 above)
Comment (yguo0905) [reply pkg/custompluginmonitor/plugin/plugin.go]: Good catch.

File: pkg/custompluginmonitor/plugin/plugin.go (hunk 2 above)
Comment (yguo0905) [reply pkg/custompluginmonitor/plugin/plugin.go]: I don't see `Stop()` is invoked anywhere. Do we support graceful shutdown?

But I changed it to use defer.

Comment (xueweiz) [reply pkg/custompluginmonitor/plugin/plugin.go]: Seems like `tomb.Stop()` is called at here:
https://github.com/kubernetes/node-problem-detector/blame/v0.6.3/pkg/custompluginmonitor/custom_plugin_monitor.go#L84

+cc @andyxning for the context, as he wrote the code.

**Comment by xueweiz** [comment]
/lgtm
/approve

This PR looks good to me. Although I probably don't have permission to approve. This is just FTR.

File: pkg/custompluginmonitor/plugin/plugin.go (hunk 2 above)
Comment (yguo0905) [reply pkg/custompluginmonitor/plugin/plugin.go]: I meant the `customPluginMonitor`'s `Stop()` doesn't seem to be used anywhere

Comment (andyxning) [reply pkg/custompluginmonitor/plugin/plugin.go]: Sorry for the late response. Actually `customPluginMonitor`'s `Stop()` is not invoked anywhere. It is just to meet the requirement about implementing `Monitor` interface. 

`Stop` should be called once `p.Run` errors and is about to exit. @yguo0905 Could you take a look at this?

**Pull request merged**